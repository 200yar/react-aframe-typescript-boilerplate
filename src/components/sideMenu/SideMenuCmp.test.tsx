import React from 'react';
import ReactDOM from 'react-dom';
import SideMenu from "./SideMenuCmp";
import renderer from 'react-test-renderer';

describe('<SideMenu/>', () => {

  it('Smoke test: renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<SideMenu />, div);
  });

  it('Matches snapshot', () => {
    const componentSnapshot = renderer.create(<SideMenu>Side Menu Content</SideMenu>).toJSON();
    expect(componentSnapshot).toMatchSnapshot();
  });

});