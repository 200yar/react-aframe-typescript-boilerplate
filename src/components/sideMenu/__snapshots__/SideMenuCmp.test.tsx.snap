// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SideMenu Component matches snapshot 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <SideMenu>
    Dummy content
</SideMenu>,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": <div
        className="sideMenuContent"
>
        <h1
                className="side-menu-title"
        />
        <hr />
        <div
                className="side-menu-btn-close"
                onClick={[Function]}
        >
                <img
                        className="vertical-center"
                        src="close-btn.svg"
                />
        </div>
        <div
                className="items-container"
        >
                <div
                        className="side-menu-item"
                >
                        Dummy content
                </div>
        </div>
        <div
                className="footer"
        >
                Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.
                <br />
                <br />
                Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10
        </div>
</div>,
      "className": "sideMenuContainer slide-in-left",
    },
    "ref": [Function],
    "rendered": Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <h1
            className="side-menu-title"
/>,
          <hr />,
          <div
            className="side-menu-btn-close"
            onClick={[Function]}
>
            <img
                        className="vertical-center"
                        src="close-btn.svg"
            />
</div>,
          <div
            className="items-container"
>
            <div
                        className="side-menu-item"
            >
                        Dummy content
            </div>
</div>,
          <div
            className="footer"
>
            Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.
            <br />
            <br />
            Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10
</div>,
        ],
        "className": "sideMenuContent",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": undefined,
            "className": "side-menu-title",
          },
          "ref": null,
          "rendered": null,
          "type": "h1",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {},
          "ref": null,
          "rendered": null,
          "type": "hr",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": <img
              className="vertical-center"
              src="close-btn.svg"
/>,
            "className": "side-menu-btn-close",
            "onClick": [Function],
          },
          "ref": null,
          "rendered": Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "className": "vertical-center",
              "src": "close-btn.svg",
            },
            "ref": null,
            "rendered": null,
            "type": "img",
          },
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <div
                className="side-menu-item"
>
                Dummy content
</div>,
            ],
            "className": "items-container",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "children": "Dummy content",
                "className": "side-menu-item",
              },
              "ref": null,
              "rendered": "Dummy content",
              "type": "div",
            },
          ],
          "type": "div",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              "Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.",
              <br />,
              <br />,
              "Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10",
            ],
            "className": "footer",
          },
          "ref": null,
          "rendered": Array [
            "Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.",
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {},
              "ref": null,
              "rendered": null,
              "type": "br",
            },
            "Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10",
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": <div
          className="sideMenuContent"
>
          <h1
                    className="side-menu-title"
          />
          <hr />
          <div
                    className="side-menu-btn-close"
                    onClick={[Function]}
          >
                    <img
                              className="vertical-center"
                              src="close-btn.svg"
                    />
          </div>
          <div
                    className="items-container"
          >
                    <div
                              className="side-menu-item"
                    >
                              Dummy content
                    </div>
          </div>
          <div
                    className="footer"
          >
                    Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.
                    <br />
                    <br />
                    Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10
          </div>
</div>,
        "className": "sideMenuContainer slide-in-left",
      },
      "ref": [Function],
      "rendered": Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            <h1
              className="side-menu-title"
/>,
            <hr />,
            <div
              className="side-menu-btn-close"
              onClick={[Function]}
>
              <img
                            className="vertical-center"
                            src="close-btn.svg"
              />
</div>,
            <div
              className="items-container"
>
              <div
                            className="side-menu-item"
              >
                            Dummy content
              </div>
</div>,
            <div
              className="footer"
>
              Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.
              <br />
              <br />
              Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10
</div>,
          ],
          "className": "sideMenuContent",
        },
        "ref": null,
        "rendered": Array [
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": undefined,
              "className": "side-menu-title",
            },
            "ref": null,
            "rendered": null,
            "type": "h1",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {},
            "ref": null,
            "rendered": null,
            "type": "hr",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": <img
                className="vertical-center"
                src="close-btn.svg"
/>,
              "className": "side-menu-btn-close",
              "onClick": [Function],
            },
            "ref": null,
            "rendered": Object {
              "instance": null,
              "key": undefined,
              "nodeType": "host",
              "props": Object {
                "className": "vertical-center",
                "src": "close-btn.svg",
              },
              "ref": null,
              "rendered": null,
              "type": "img",
            },
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                undefined,
                <div
                  className="side-menu-item"
>
                  Dummy content
</div>,
              ],
              "className": "items-container",
            },
            "ref": null,
            "rendered": Array [
              undefined,
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {
                  "children": "Dummy content",
                  "className": "side-menu-item",
                },
                "ref": null,
                "rendered": "Dummy content",
                "type": "div",
              },
            ],
            "type": "div",
          },
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "host",
            "props": Object {
              "children": Array [
                "Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.",
                <br />,
                <br />,
                "Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10",
              ],
              "className": "footer",
            },
            "ref": null,
            "rendered": Array [
              "Note: this is cutting-edge software. It can be unstable and contain bugs due to third party code.",
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              Object {
                "instance": null,
                "key": undefined,
                "nodeType": "host",
                "props": Object {},
                "ref": null,
                "rendered": null,
                "type": "br",
              },
              "Tested in mobile Chorme 66, Destokp Chrome 66 and Windows 10",
            ],
            "type": "div",
          },
        ],
        "type": "div",
      },
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
